samcli/lib/telemetry/metric.py

```diff
@@ -124,21 +124,28 @@ def hello_command():
 
     """
 
+    @wraps(func)
     def wrapped(*args, **kwargs):
-        telemetry = Telemetry()
-
         exception = None
         return_value = None
         exit_reason = "success"
         exit_code = 0
 
         duration_fn = _timer()
+
+        ctx = None
+        try:
+            ctx = Context.get_current_context()
+        except RuntimeError:
+            LOG.debug("Unable to find Click Context for getting session_id.")
```

Comment:
This was just me moving stuff around, so this was the same from before. As for why this exception is being swallowed, I am not entirely sure. If we don't have access to the click context, then we don't send metrics since we rely on that click context to grab things like command names and template data.